### PR TITLE
Feature/feedback

### PR DESCRIPTION
**Pull request by kylejmorris**
design of feedback mechanism: mimics mixpanel.track design. We can give this to Bounte and other customers so they can track what users think about inference generations. 
```
JS call:
banana.feedback(apiKey, callID, feedback = {})
feedback payload:
feedback = {
   metric: "upvote/downvote/favorited/copied/engagement/etc"
   properties: {
      // any content the user wants to save as part of this feedback. comments, metadata, upvote/downvote, star ratings, etc
   } 
}
```

Tested locally with a stub backend. Summer also works. I return CallID from banana.run now so we can immediately give it to customer to provide feedback.

### CODE REVIEW

File: module/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "dependencies": {
     "axios": "^0.21.0",
     "image-to-base64": "^2.1.1",
+    "typescript": "^4.5.5",
```
Comment (erik-dunteman) [inline module/package.json]: package itself theoretically doesn't need typescript dependency

File: module/genericsUtils.ts [hunk 2]
```diff
@@ -70,6 +76,34 @@ const startAPI = async (apiKey: string, modelKey: string, modelInputs: object, s
     return callID
 }
 
+const feedbackAPI = async (apiKey: string, callID: string, feedback: object): Promise<any> => {
+    const url = endpoint.concat("feedback/")
```
Comment (erik-dunteman) [inline module/genericsUtils.ts]: something I forgot to mention in middleware is a small nitpick: should we do this as feedback/v1 instead? I guess we control it and could be introducing unnecessary slowness in our dev process by doing these nitpicks, but it's a thought.

File: module/genericsUtils.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ export async function runMain(apiKey: string, modelKey: string, modelInputs: obj
         const jsonOut = await checkAPI(apiKey, callID)
         if (jsonOut !== undefined){
             if (jsonOut.message.toLowerCase() === "success"){
+                jsonOut['CallID'] = callID
```
Comment (erik-dunteman) [inline module/genericsUtils.ts]: we fixed this together, should be callID

**Review by erik-dunteman — CHANGES_REQUESTED** [review]
Not approved - you've deleted (or gitignored) the dist directory, which is actually what's used on npm publish and import. Please reinclude those compiled .js and .d.ts files

**Comment by kylejmorris** [comment]
added

**Review by erik-dunteman — APPROVED** [review]
yeeeeeee haw

**Pull request merged**